odl/space/fspace.py

```diff
@@ -379,14 +395,6 @@ def __call__(self, x, out=None, **kwargs):
             if ndim == 1:
                 try:
                     out = self._call(x, **kwargs)
-                    if np.ndim(out) == 0 and not scalar_out:
-                        # Don't accept scalar result. A typical situation where
-                        # this occurs is with comparison operators, e.g.
-                        # "return x > 0" which simply gives 'True' for a
-                        # non-empty tuple (in Python 2). We raise TypeError
-                        # to trigger the call with x[0].
```

Comment:
Not quite sure to be frank, this piece of code is a rather deep rabbit hole.

Apparently it is [here](https://github.com/odlgroup/odl/blob/60e2af8b790877dfe7fc04d2565d24e48eb88655/odl/space/fspace.py#L363-L365).